server/tests/api_v4/authenticated/test_realm_rename.py

```diff
@@ -294,28 +272,95 @@ async def test_authenticated_realm_rename_bad_key_index(
     )
 
 
-@pytest.mark.parametrize("kind", ("dummy_data", "bad_author"))
+@pytest.mark.parametrize(
+    "kind",
+    ("dummy_data", "bad_author"),
+)
 async def test_authenticated_realm_rename_invalid_certificate(
-    coolorg: CoolorgRpcClients, backend: Backend, kind: str
+    coolorg: CoolorgRpcClients, kind: str
 ) -> None:
-    t1 = DateTime.now()
     match kind:
         case "dummy_data":
-            realm_key_rotation_certificate = b"<dummy data>"
+            certif = b"<dummy data>"
         case "bad_author":
-            certif = RealmNameCertificate(
-                author=coolorg.bob.device_id,
-                timestamp=t1,
-                realm_id=coolorg.wksp1_id,
-                key_index=1,
-                encrypted_name=b"<encrypted name>",
+            certif = realm_name_certificate(coolorg.bob, realm_id=coolorg.wksp1_id).dump_and_sign(
+                coolorg.bob.signing_key
             )
-            realm_key_rotation_certificate = certif.dump_and_sign(coolorg.bob.signing_key)
-        case _:
-            assert False
+        case unknown:
+            assert False, unknown
 
     rep = await coolorg.alice.realm_rename(
-        realm_name_certificate=realm_key_rotation_certificate,
+        realm_name_certificate=certif,
         initial_name_or_fail=False,
     )
     assert rep == authenticated_cmds.v4.realm_rename.RepInvalidCertificate()
+
+
+@pytest.mark.xfail(
+    reason="TODO: realm_rename never returns RepRequireGreaterTimestamp. Not implemented?"
+)
+@pytest.mark.parametrize(
+    "timestamp_offset",
+    (pytest.param(0, id="same_timestamp"), pytest.param(1, id="previous_timestamp")),
+)
+async def test_authenticated_realm_rename_require_greater_timestamp(
```

Comment:
Not sure if there is a problem with the test or the implementation is missing.

Reply:
good catch ! this is an error in the implementation